src/main/java/au/org/emii/ncdfgenerator/NcdfEncoderBuilder.java

```diff
@@ -0,0 +1,1476 @@
+
+
+package au.org.emii.ncdfgenerator;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream ;
+import java.io.OutputStream ;
+import java.io.IOException;
+
+import java.nio.charset.Charset;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern ;
+import java.util.regex.Matcher;
+import java.util.AbstractMap.SimpleImmutableEntry;
+
+import java.sql.*;
+
+import java.text.SimpleDateFormat;
+
+import ucar.nc2.NetcdfFileWriteable;
```

Comment:
Yes, we'll need to update to encode the latest ncdf version. I attempted to use it, but ran into some issue I can't remember now (weird writable factory instantiation?), and so fell back on using the original library as a prototype.